### PR TITLE
test: add edge case tests for useCountdown

### PR DESCRIPTION
**Pull request by snowbitx**
### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [ ] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

add edge case tests for useCountdown

### CODE REVIEW

File: packages/core/useCountdown/index.test.ts [hunk 1]
```diff
@@ -142,4 +142,19 @@ describe('useCountdown', () => {
...
+    start()
+    start()
+    start()
+    vi.advanceTimersByTime(310)
+    expect(tickCallback).toHaveBeenCalledTimes(3)
+  })
+
+  it('isActive becomes false after countdown completes', async () => {
+    const { isActive } = useCountdown(2, options)
+    vi.advanceTimersByTime(210)
```
Comment (OrbisK) [inline packages/core/useCountdown/index.test.ts]: can we check if it is true before advancing the timers? 😄

File: packages/core/useCountdown/index.test.ts [hunk 2]
```diff
@@ -142,4 +142,19 @@ describe('useCountdown', () => {
...
     expect(completeCallback).toHaveBeenCalledTimes(3)
   })
+
+  it('multiple start() calls only create one timer', async () => {
+    const { start } = useCountdown(3, options)
+    start()
+    start()
+    start()
+    vi.advanceTimersByTime(310)
+    expect(tickCallback).toHaveBeenCalledTimes(3)
```
Comment (OrbisK) [inline packages/core/useCountdown/index.test.ts]: why 310? should'nt we advance by +1000ms?

**Review by OrbisK — CHANGES_REQUESTED** [review]

File: packages/core/useCountdown/index.test.ts (hunk 2 above)
Comment (snowbitx) [reply packages/core/useCountdown/index.test.ts]: > why 310? should'nt we advance by +1000ms?

Good catch — I originally used 310ms by referencing the 'start can provide a custom interval' test case (which also uses short intervals), but you’re right that 1000ms makes it much clearer. I've updated it accordingly — thanks! 🙌

File: packages/core/useCountdown/index.test.ts (hunk 1 above)
Comment (snowbitx) [reply packages/core/useCountdown/index.test.ts]: > can we check if it is true before advancing the timers? 😄

Thanks for the suggestion — added the initial check for isActive.

File: packages/core/useCountdown/index.test.ts [hunk 3]
```diff
@@ -142,4 +142,20 @@ describe('useCountdown', () => {
...
+  it('multiple start() calls only create one timer', async () => {
+    const { start } = useCountdown(3, options)
+    start()
+    start()
+    start()
+    vi.advanceTimersByTime(1000)
+    expect(tickCallback).toHaveBeenCalledTimes(3)
+  })
+
+  it('isActive becomes false after countdown completes', async () => {
```
Comment (Alfred-Skyblue) [inline packages/core/useCountdown/index.test.ts:155]: This feature has corresponding test cases.
https://github.com/vueuse/vueuse/blob/d23812dbebece77e5e0bd6d9d3574a3637ac31b0/packages/core/useCountdown/index.test.ts#L58-L67

Comment (snowbitx) [reply packages/core/useCountdown/index.test.ts:155]: > This feature has corresponding test cases.
> 
> https://github.com/vueuse/vueuse/blob/d23812dbebece77e5e0bd6d9d3574a3637ac31b0/packages/core/useCountdown/index.test.ts#L58-L67

Confirmed — existing tests already cover it. Sorry for the confusion and thanks! 🙏
